lib/spack/spack/operating_systems/windows_os.py

```diff
@@ -65,8 +69,35 @@ class WindowsOs(OperatingSystem):
                 )
         except (subprocess.CalledProcessError, OSError, UnicodeDecodeError):
             pass
-    if comp_search_paths:
-        compiler_search_paths = comp_search_paths
+    _compiler_search_paths.extend(comp_search_paths)
```

Comment:
I think this will automatically be resolve as part of my comment about only doing this for `__init__` but because these variable definitions all live at class instantiation level, it's hard to track what the "real" variable-that-stores-the-result-of-the-computation-and-thing-you-want-to-ultimately-use is.

Reply:
I think you're right, resolving the comment above does address this, but I also cut down on the number of intermediate variables we're using.